pyomo/contrib/viewer/qt.py

```diff
@@ -40,18 +40,24 @@ class DummyQAbstractItemModel(object):
     def __init__(*args, **kwargs):
         pass
 
+class DummyQAbstractTableModel(object):
+    """
+    A dummy QAbstractTableModel class to allow some testing without PyQt
+    """
+    def __init__(*args, **kwargs):
+        pass
+
 qt_available = False
+qt_import_errors = []
 
 try:
     from PyQt5 import QtCore
 except:
-    _log.exception("Cannot import PyQt5.QtCore")
+    qt_import_errors.append("Cannot import PyQt5.QtCore")
     try:
         from PyQt4 import QtCore
     except:
-        _log.exception("Cannot import PyQt4.QtCore")
-        QAbstractItemModel = DummyQAbstractItemModel
-        QtCore = DummyQtCore
+        qt_import_errors.append("Cannot import PyQt4.QtCore")
```

Comment:
Don't quite understand why we capture the errors to report later instead of delaying the import until it is needed? Can you clarify?

Reply:
I was working to minimally change @eslickj's code.

I think his motivation was to enable testing of a good chunk of the viewer interface without PyQt. So some of the framework (including the Qt wrappers) need to be imported during testing. It became an issue for us when we asked to move the executable that drives the UI out of console_scripts and into a Pyomo subcommand.  This causes Pyomo to have to import more of the viewer than it used to when users run the pyomo command, so pyqt import error messages were cropping up (e.g., in unrelated tests).